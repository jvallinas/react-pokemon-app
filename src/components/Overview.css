@import url('https://fonts.googleapis.com/css?family=Big+Shoulders+Text&display=swap');

.Overview {
  text-align: initial;
}

.header-container {
    height: 400px;
    font-size: 2rem;
    background-color: rgba(46, 26, 4, 1);
    display: flex;
    flex-direction: column;
}

.header-container > * {
    font-family: 'Big Shoulders Text', Arial, Helvetica, sans-serif, sans-serif;
    color: white;
    text-align: center;
}

.header-container > .header-title {
    text-shadow: 2px 2px 5px lightblue;
}

button.go-to-selection {
    display: block;
    border: 0;
    border-radius: 8px;
    font-size: 1em;
    padding: 1em 3em;
    margin: 0 auto;
    margin-bottom: 48px;
    font-family: 'Big Shoulders Text', Arial, Helvetica, sans-serif, sans-serif;
    background-color: purple;
    color: white;
}

.header-container > button.go-to-selection {
    font-size: 0.5em;
    margin-top: auto;
}

h2.section-title, .description {
    color: black;
    font-family: 'Big Shoulders Text', cursive, sans-serif;
}

section.description {
    padding: 16px 0;
    margin: 0 10%;
}