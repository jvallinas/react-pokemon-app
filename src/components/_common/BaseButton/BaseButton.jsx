import React from 'react';
import PropTypes from 'prop-types';
import styles from './BaseButton.module.css';

const baseButtonPropTypes = {
  label: PropTypes.string.isRequired,
  onClickHandler: PropTypes.func.isRequired,
  styleOptions: PropTypes.arrayOf(PropTypes.string),
};

const BaseButton = ({ label, onClickHandler, styleOptions }) => {
  const mappedStyleClasses = styleOptions.map((option) => styles[option]).join(' ');
  return (
    <>
      <button
        type="button"
        className={mappedStyleClasses}
        onClick={onClickHandler}
      >
        {label}
      </button>
    </>
  );
};

BaseButton.defaultProps = {
  styleOptions: [''],
};
BaseButton.propTypes = baseButtonPropTypes;

export default React.memo(BaseButton);
