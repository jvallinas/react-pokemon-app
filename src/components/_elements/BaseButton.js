import React from 'react';
import PropTypes from 'prop-types';
import styles from './BaseButton.module.css';

const baseButtonPropTypes = {
  label: PropTypes.string.isRequired,
  onClickHandler: PropTypes.func.isRequired,

  // Style props
  uppercase: PropTypes.bool,
  purple: PropTypes.bool
}

const BaseButton = ({ label, onClickHandler, ...styleProps }) => {
  
  const stylesList = styleProps && Object.keys(styleProps).map(c => styles[c]).join(' ');
  return (
    <>
      <button 
        className={stylesList}
        onClick={onClickHandler} 
      >{label}</button>
    </>
  );
}

BaseButton.propTypes = baseButtonPropTypes;

export default React.memo(BaseButton);