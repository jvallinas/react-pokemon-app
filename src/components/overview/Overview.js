import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './Overview.module.css';

import OverviewSteps, {STEPS_INFO} from './OverviewSteps';

const overviewPropTypes = {
	title: PropTypes.string,
	subtitle: PropTypes.string
}

function Overview(props) {
	useEffect(() => {
		document.title = `Welcome to the React Pokemon App`;
	}, []);

	let history = useHistory();
	const navigateToSelection = () => history.push('/selection');

	return (
		<>
			<div className={styles['header-container']}>
				<h1 className={styles['header-title']}>{props.title.toUpperCase()}</h1>
				<div className={styles['header-subtitle']}>{props.subtitle.toUpperCase()}</div>
				<button className={styles['go-to-selection']} onClick={navigateToSelection}>{`LET'S DO IT!`}</button>
			</div>

			<div className={styles['feature-list']}>
				<section>
					<h1 className={styles['section-title']}>OBJECTIVE</h1>
					<div className={styles['description']}>The goal is to create a Pokemon application using the React framework and the Pokemon API.</div>
				</section>
				<section>
					<h1 className={styles['section-title']}>CHALLENGE</h1>
					<div className={styles['description']}>
						We will be creating a model to instantiate new pokemons and supply our application with data, having several fields such as their name, type, weakness and other stats.
				</div>
				</section>
				<section>
					<h1 className={styles['section-title']}>HOW IT WORKS</h1>
					<video width="auto" height="auto" controls>
						(Your browser does not support the video tag).
				</video>
				</section>
			</div>

			<div className={styles['steps']}>
				{STEPS_INFO.map((step, index) => 
					<OverviewSteps 
						key={index} 
						stepTitle={`STEP ${++index}`} 
						stepDescription={step.description}
						imagePath={step.imagePath}
					/>
				)}
			</div>

			<button className={styles['go-to-selection']} onClick={navigateToSelection}>{`LET'S DO IT!`}</button>
		</>
	);
}

Overview.propTypes = overviewPropTypes;

export default Overview;