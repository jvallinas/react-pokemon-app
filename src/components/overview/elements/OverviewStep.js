import React from 'react';
import PropTypes from 'prop-types';
import styles from './OverviewStep.module.css';

const overviewStepPropTypes = {
	imagePath: PropTypes.string,
	stepTitle: PropTypes.string,
	stepDescription: PropTypes.string
}

const OverviewStep = ({imagePath, stepTitle, stepDescription}) => {
	return (
		<>
			<div className={styles['step-container']}>
				<img className={styles['step-image-container']} src={imagePath} alt=""></img>
				<h1 className={styles['step-title']}>{stepTitle}</h1>
				<div className={styles['step-description']}>{stepDescription}</div>
			</div>
		</>
	);
}

OverviewStep.propTypes = overviewStepPropTypes;

export default OverviewStep;