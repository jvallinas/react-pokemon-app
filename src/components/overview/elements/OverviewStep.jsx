import React from 'react';
import PropTypes from 'prop-types';
import styles from './OverviewStep.module.css';

const overviewStepPropTypes = {
  imagePath: PropTypes.string.isRequired,
  stepTitle: PropTypes.string.isRequired,
  stepDescription: PropTypes.string.isRequired,
};

const OverviewStep = ({ imagePath, stepTitle, stepDescription }) => (
  <>
    <div className={styles['step-container']}>
      <img className={styles['step-image-container']} src={imagePath} alt="" />
      <h1 className={styles['step-title']}>{stepTitle}</h1>
      <div className={styles['step-description']}>{stepDescription}</div>
    </div>
  </>
);

OverviewStep.propTypes = overviewStepPropTypes;

export default OverviewStep;
