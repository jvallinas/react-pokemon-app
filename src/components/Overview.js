import React, { useEffect } from 'react';
import './Overview.css';

function Overview() {
	useEffect(() => {
		document.title = `Welcome to the React Pokemon App`;
	}, []);

	return (
		<div className="Overview">
			<div className="header-container">
				<h1 className="header-title">POKEMON BATTLE</h1>
				<div className="subtitle">Test your skills building a Pokemon application</div>
				<button className="go-to-selection">{`LET'S DO IT!`}</button>
			</div>

			<section className="description">
				<h2 className="section-title">OBJECTIVE</h2>
				<div className="description">The goal is to create a Pokemon application using the React framework and the Pokemon API.</div>

				<h2 className="section-title">CHALLENGE</h2>
				<div className="description">
					We will be creating a model to instantiate new pokemons and supply our application with data, having several fields such as their name, type, weakness and other stats.
        </div>

				<h2 className="section-title">HOW IT WORKS</h2>
				<video width="auto" height="auto" controls>
					(Your browser does not support the video tag).
        </video>
			</section>

			<button className="go-to-selection">{`LET'S DO IT!`}</button>
		</div>
	);
}

export default Overview;