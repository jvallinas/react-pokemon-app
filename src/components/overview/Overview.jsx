import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './Overview.module.css';

import BaseButton from '../_elements/BaseButton';
import OverviewStep from './elements/OverviewStep';

const overviewPropTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

const STEPS_INFO = [
  {
    id: 1,
    imagePath: 'images/pachirisu.png',
    description: 'Info for the first step',
  },
  {
    id: 2,
    imagePath: 'images/pachirisu.png',
    description: 'Info for the second step',
  },
  {
    id: 3,
    imagePath: 'images/pachirisu.png',
    description: 'Info for the third and final step',
  },
];

const Overview = ({ title, subtitle }) => {
  useEffect(() => {
    document.title = title;
  }, [title]);

  const history = useHistory();
  const navigateToSelection = () => history.push('/selection');

  return (
    <>
      <div className={styles['header-container']}>
        <h1 className={styles['header-title']}>{title.toUpperCase()}</h1>
        <div className={styles['header-subtitle']}>{subtitle.toUpperCase()}</div>
        <BaseButton label={'Let\'s do it!'} onClickHandler={navigateToSelection} uppercase purple />
      </div>

      <div className={styles['feature-list']}>
        <section>
          <h1 className={styles['section-title']}>OBJECTIVE</h1>
          <div className={styles.description}>
            {`The goal is to create a Pokemon application 
						using the React framework and the Pokemon API.`}
          </div>
        </section>
        <section>
          <h1 className={styles['section-title']}>CHALLENGE</h1>
          <div className={styles.description}>
            We will be creating a model to instantiate new pokemons
						and supply our application with data,
						having several fields such as their name, type, weakness and other stats.
          </div>
        </section>
        <section>
          <h1 className={styles['section-title']}>HOW IT WORKS</h1>
          <video width="auto" height="auto" controls>
            <track kind="captions" />

						(Your browser does not support the video tag).
          </video>
        </section>
      </div>

      <div className={styles.steps}>
        {STEPS_INFO.map((step, index) => (
          <OverviewStep
            key={step.id}
            stepTitle={`STEP ${index + 1}`}
            stepDescription={step.description}
            imagePath={step.imagePath}
          />
        ))}
      </div>

      <BaseButton label={'Let\'s do it!'} onClickHandler={navigateToSelection} uppercase purple />
    </>
  );
};

Overview.propTypes = overviewPropTypes;

export default Overview;
