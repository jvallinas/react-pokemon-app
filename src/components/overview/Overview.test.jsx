import React from 'react';
import { shallow } from 'enzyme';
import Overview from './Overview';
import OverviewStep from './elements/OverviewStep';
import BaseButton from '../_common/BaseButton/BaseButton';

// Workaround for useHistory since react-router context isn't set in this component
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

const mockTitle = 'Welcome to React Pokemon app';
const mockSubtitle = 'An app for learning React';

describe('Layout rendering in Overview page', () => {
  it('Renders correct number of button elements', () => {
    const wrapper = shallow(<Overview title={mockTitle} subtitle={mockSubtitle} />);
    expect(wrapper.find(BaseButton).length).toEqual(2);
  });

  it('Renders correct number of Step items', () => {
    const wrapper = shallow(<Overview title={mockTitle} subtitle={mockSubtitle} />);
    expect(wrapper.find(OverviewStep).length).toEqual(3);
  });
});
