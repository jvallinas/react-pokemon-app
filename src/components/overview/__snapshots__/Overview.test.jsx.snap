// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout rendering in Overview page Renders as expected 1`] = `
Array [
  <div
    className="header-container"
  >
    <h1
      className="header-title"
    >
      WELCOME TO REACT POKEMON APP
    </h1>
    <div
      className="header-subtitle"
    >
      AN APP FOR LEARNING REACT
    </div>
    <button
      className="uppercase purple"
      onClick={[Function]}
      type="button"
    >
      Let's do it!
    </button>
  </div>,
  <div
    className="feature-list"
  >
    <section>
      <img
        alt=""
        className="description-decoration"
        height="400"
        src="../images/pachirisu.png"
        width="auto"
      />
      <h1
        className="section-title"
      >
        OBJECTIVE
      </h1>
      <div
        className="description"
      >
        The goal is to create a Pokemon application using the React framework and the Pokemon API. React Hooks and functional components will be used for this app.
      </div>
    </section>
    <section>
      <h1
        className="section-title"
      >
        CHALLENGE
      </h1>
      <div
        className="description"
      >
        We will be creating a model to instantiate new pokemons and supply our application with data, having several fields such as their name, type, weakness and other stats. This information will be stored via Redux store, dispatching different actions based on the type of info to store.
      </div>
    </section>
    <section>
      <h1
        className="section-title"
      >
        HOW IT WORKS
      </h1>
      <video
        controls={true}
        height="auto"
        min-width="300px"
        width="80%"
      >
        <source
          src="../sample_video.mp4"
          type="video/mp4"
        />
        <track
          kind="captions"
        />
        (Your browser does not support the video tag).
      </video>
    </section>
  </div>,
  <div
    className="steps"
  >
    <div
      className="step-container"
    >
      <img
        alt=""
        className="step-image-container"
        src="images/pokeball.png"
      />
      <h1
        className="step-title"
      >
        STEP 1
      </h1>
      <div
        className="step-description"
      >
        Go to the Selection screen, where you will be able to select from a list of many different Pokemons
    and choose them for battle. It is also possible to filter pokemons by their name, or by their type.
      </div>
    </div>
    <div
      className="step-container"
    >
      <img
        alt=""
        className="step-image-container"
        src="images/pokedex.png"
      />
      <h1
        className="step-title"
      >
        STEP 2
      </h1>
      <div
        className="step-description"
      >
        By selecting a Pokemon, their full stats will be displayed so that you can confirm if you want it to take part in a battle.
    If you change your mind, you can always cancel by closing the dialog and select another Pokemon from the Selection screen.
      </div>
    </div>
    <div
      className="step-container"
    >
      <img
        alt=""
        className="step-image-container"
        src="images/battle.png"
      />
      <h1
        className="step-title"
      >
        STEP 3
      </h1>
      <div
        className="step-description"
      >
        When two Pokemons are selected, gather them together and go to the Fight screen in order for the battle to start!
      </div>
    </div>
  </div>,
  <button
    className="uppercase purple"
    onClick={[Function]}
    type="button"
  >
    Let's do it!
  </button>,
]
`;
